src/kaboodle.rs

```diff
@@ -742,3 +748,33 @@ impl KaboodleInner {
         }
     }
 }
+
+fn calculate_peer_latency(peer_info: &PeerInfo) -> Option<Duration> {
+    let ping_send_time = match peer_info.state {
+        PeerState::WaitingForIndirectPing(ping_send_time)
+        | PeerState::WaitingForPing(ping_send_time) => ping_send_time,
+        _ => {
+            // We weren't expecting to hear from this peer, so we have no start time to compare
+            // against; just return the previously known latency value, if there is one.
+            return peer_info.latency;
+        }
+    };
+
+    let ping_latency = Instant::now()
+        .checked_duration_since(ping_send_time)
+        .unwrap();
+    let Some(prev_latency) = peer_info.latency else {
+        // No previous latency, so just return this new value directly
+        return Some(ping_latency);
+    };
+
+    // Mix the new latency into the old one, giving 80% weight to the new value; this should smooth
+    // out any wild swings while remaining fairly responsive to changing conditions.
+    const MOST_RECENT_WEIGHT: f64 = 0.8;
```

Comment:
this could be defined up top? Feels surprising to see a random official-looking const pop up half-way through the body of a function.

Reply:
I generally only hoist constants up if they're either used in multiple places or are really key pieces of information (see, for example, all of the constants at the top of kaboodle.rs that control the mesh discovery stuff). For little things like this, I usually leave 'em in the function body; I am pretty sure the compiler is just going to inline that as a literal anyhow.